index.js

```diff
@@ -1,183 +1,187 @@
     if (this.options.unwatchOnRelease) {
-        resource.UNWATCH();
+      resource.UNWATCH();
     }
 
-    var pool = this.pools[database];
+    const pool = this.pools[database];
 
     if (pool) {
-        pool.release(resource);
+      await pool.release(resource);
     }
-};
+  }
 
+  _addCommands () {
+    if (this.options.commands.length) {
+      this.options.commands.forEach(newCommand => redis.add_command(newCommand));
+    }
+  }
+
+  _emitStatus() {
+    setInterval(() => {
+      for (const [poolKey, pool] of Object.entries(this.pools)) {
+        this.emit('status', {
+          name: this.options.name,
+          db: poolKey,
+          count: pool.size,
+          unused: pool.available,
+          waiting: pool.pending
+        });
+      }
+    }, this.options.emitter.statusInterval);
+  }
+};
 
 /**
  * Factory to create new Redis pools for a given Redis database
  * @param options
  * @param database
- * @returns {Object}
+ * @returns {Pool}
  */
-function makePool(options, database) {
-    return Pool({
-        name: options.name + ':' + database,
-
-        create: function(callback) {
-
-            var callbackCalled = false;
-
-            var client = redis.createClient(options.port, options.host, {
-                no_ready_check: options.noReadyCheck
-            });
-
-            client.on('error', function (err) {
-                log(options, {db: database, action: 'error', err: err.message});
-                if (!callbackCalled) {
-                    callbackCalled = true;
-                    callback(err, client);
-                }
-                client.end(FLUSH_CONNECTION);
-            });
-
-            client.on('ready', function () {
-                client.select(database, function(err/*, res*/) {
-                    if (!callbackCalled) {
-                        callbackCalled = true;
-                        callback(err, client);
-                    }
-                });
-            });
-        },
-
-        destroy: function(client) {
-            client.quit();
+function makePool (options, database) {
+  const factory = {
+    create () {
+      return new Promise((resolve, reject) => {
+        let settled = false;
+
+        const client = redis.createClient(options.port, options.host, {
+          no_ready_check: options.noReadyCheck
+        });
+
+        client.on('error', function (err) {
+          log(options, { db: database, action: 'error', err: err.message });
+
+          if (!settled) {
+            settled = true;
             client.end(FLUSH_CONNECTION);
-        },
-
-        validate: function(client) {
-            return client && client.connected;
-        },
-
-        max: options.max,
-        idleTimeoutMillis: options.idleTimeoutMillis,
-        reapIntervalMillis: options.reapIntervalMillis,
-        returnToHead: options.returnToHead,
-        log: options.log
-    });
-}
 
+            if (err) {
+              return reject(err);
```

Comment:
Careful here:  
https://github.com/coopernurse/node-pool/issues/175
https://github.com/coopernurse/node-pool/issues/183

It happened to me in Cartonik: https://github.com/CartoDB/cartonik/blob/master/lib/renderer/map-pool.js#L23-L43